### PR TITLE
cleanup the fixed bitwidth compressor

### PR DESCRIPTION
**Pull request by kedartatwawadi**
1. cleanup
2. create an encoder which uses pickle for alphabet encoding/decoding
3. fix the func which returns a list from numpy array, as if it is not a list, the pickled side is ~10x larger

### CODE REVIEW

File: external_compressors/pickle_external.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        length_encoding = uint_to_bitarray(len_pickled, bit_width=self.length_bitwidth)
+        return length_encoding + pickled_bits
+
+
+class PickleDecoder(DataDecoder):
+    def __init__(self, length_bitwidth=32):
+        self.length_bitwidth = length_bitwidth
+
+    def decode_block(self, bitarray: BitArray):
+        length_bitwidth = 32
```
Comment (shubhamchandak94) [inline external_compressors/pickle_external.py:40]: why set to 32 here and not use self.length_bitwidth?

File: compressors/fixed_bitwidth_compressor.py [hunk 2]
```diff
@@ -161,7 +185,23 @@ def test_fixed_bitwidth_encode_decode():
     assert codelen == len(data_list) * 2 + alphabet_bits
 
 
-def test_fixed_bitwidth_file_encode_decode():
+def test_fixed_bitwidth_encode_decode():
```
Comment (shubhamchandak94) [inline compressors/fixed_bitwidth_compressor.py:188]: test also non-character alphabet?

File: external_compressors/pickle_external.py (hunk 1 above)
Comment (kedartatwawadi) [reply external_compressors/pickle_external.py:40]: thanks! ll fix

File: compressors/fixed_bitwidth_compressor.py (hunk 2 above)
Comment (kedartatwawadi) [reply compressors/fixed_bitwidth_compressor.py:188]: as in? can you elaborate?

Comment (shubhamchandak94) [reply compressors/fixed_bitwidth_compressor.py:188]: Just meant test on integer or something.

**Review by shubhamchandak94 — APPROVED** [review]

**Pull request merged**